contents/handbook/people/team-structure/pipeline/objectives.mdx

```diff
@@ -0,0 +1,24 @@
+- Objective: Events are ingested and processed reliably, accurately, and quickly on all deployments
+    - Key result:
+        - End-to-end P95 time to ingest events is under 1m30s for "normal events" and 2m30s for "buffer events" (note: includes 60s error bars)
+        - We ingest 99.999% of valid events
+            - Limit dead letter queue usage
+        - Ingestion on self-hosted deployments is horizontally scalable
```

Comment:
how do you measure this?

Reply:
So horizontal scalability is the ability to add more servers and have the work be distributed appropriately across them so that throughput/performance of the system/service overall improves. I believe you may already have known that, but covering it just in case. 

Currently, this is pretty clear cut - if we add an additional plugin server to a self-hosted installation, that server won't do anything because of some configuration we have (that's tricky to update). In even more concrete terms, if you look at its logs and metrics, you will see that it is not processing any events. The goal is achieved if new servers added actually do some work. This can be verified with logs, metrics, etc. 

Hope that answers it! Great question

Reply:
Thanks for the additional context here! So success for this key result would be adding a new plugin server and getting it to a place where it can actually process events.

Do you expect that you'll be able to verify those output metrics this quarter? Is there a key metric you could aim to improve that would make this KR measurable?

Reply:
>  So success for this key result would be adding a new plugin server and getting it to a place where it can actually process events.

yes, if assuming by "adding a new plugin server" you mean if a self-hosted user tries to use >1 replicas for the plugin server. Note that on cloud we already are using multiple replicas.

> Do you expect that you'll be able to verify those output metrics this quarter? Is there a key metric you could aim to improve that would make this KR measurable?

This is very easily measurable, it works or not (currently) and we can just test it with spinning up a self-hosted deployment or just on playground where we test releases.

---
With regards to the others measurability: 
1. End-to-end P95 time to ingest events - we hope to not need to do any work here, but it's important, so if our scale or other changes push that metric too far it will be something we'd prioritise fixing over new features, ... Current graphs: https://metrics.posthog.com/d/CD1kWjLnz/plugin-server-new?orgId=1&from=now-14d&to=now&viewPanel=83 & https://metrics.posthog.com/d/CD1kWjLnz/plugin-server-new?orgId=1&from=now-14d&to=now&viewPanel=84
2. We ingest 99.999% of valid events - https://posthog.slack.com/archives/C0374DA782U/p1664831916352319